src/codewhisperer/explorer/codewhispererNode.ts

```diff
@@ -55,9 +55,13 @@ export class CodeWhispererNode implements RootNode {
 
     private getDescription(): string {
         if (AuthUtil.instance.isConnectionValid()) {
-            return AuthUtil.instance.isEnterpriseSsoInUse()
-                ? 'IAM Identity Center Connected'
-                : 'AWS Builder ID Connected'
+            if (AuthUtil.instance.isEnterpriseSsoInUse()) {
+                return 'IAM Identity Center Connected'
+            } else if (AuthUtil.instance.isBuilderIdInUse()) {
+                return 'AWS Builder ID Connected'
+            } else {
+                return 'IAM Connected'
+            }
```

Comment:
what's the logic for `isConnectionValid` when it's for SageMaker IAM? I don't see it's being changed. Does it work out of the box?

Reply:
`isConnectionValid` just checks for a live, non-expired, connection. So this could be Builder ID, SSO, or IAM. So we get it out of the box